scripts/cesm3_LTso_workflow_TEMPLATE.sh

```diff
@@ -0,0 +1,338 @@
+casedir=/glade/derecho/scratch/nanr/workflow_testcase/
+rundir=/glade/derecho/scratch/nanr/workflow_testrun/
+
+CODE_ROOT=/glade/derecho/scratch/nanr/cesm_tags/
+CASENAME=b.e30_${cesmtag:(-8)}.${compset}.${useresoln}.cmip7-testing.006
+CASEROOT=$casedir/$CASENAME
+
+echo "creating $CASENAME"
+
+########################
+## Set flags to do stuff ----
+########################
+do_sudox=false
+do_git_archive=false
+do_download_code=false
+do_create_newcase=true
+do_case_setup=true
+do_case_build=false
+do_case_submit=false
+do_tseries=false
+do_cupid=false
+do_cmor=false
+
+########################
+## Set atm output frequency
+########################
+do_hr3=true
+do_hr6=true
+do_dy=true
+
+########################
+## connect as cmip7 user
+########################
+if [ "${do_sudox,,}" == "true" ]; then
+     echo $'\n----- Switching to CMIP7 user ID -----\n'
+     sudox cmip7
+     whoami
+fi
+
+########################
+## Set Case Details
+########################
+
+########################
+## Clone the repository
+########################
+
+if [[ $do_download_code != true ]]; then
+     echo $'\n----- Skipping code download -----\n'
+else
+
+     echo $'\n----- Downloading CESMROOT -----\n'
+     path=${CODE_ROOT}/${cesmtag}
+     if [ -d "${path}" ]; then
+        echo "ERROR: CESMROOT Directory already exists. Not overwriting"
+        exit 20
+     fi
+
+     cd $path
+
+     # clone the repo and get the cesm tag
+     git clone https://github.com/ESCOMP/cesm $cesmtag -b $cesmtag
+     cd $cesmtag
+
+     # check out the components
+     ./bin/git-fleximod update
+
+     ###############################################
+     ## Update the ccs_config to the workflow branch
+     ###############################################
+     cd ccs_config
+     git checkout add_cmip7_workflow_amon_basic
+
+fi
+
+########################
+## Create new case
+########################
+if [[ $do_create_newcase != true ]]; then
+     echo $'\n----- Skipping create_newcase -----\n'
+else
+    path=${CASEROOT}
+    if [ -d "${path}" ]; then
+        echo "ERROR: CASE Directory already exists. Not overwriting"
+        exit 20
+    fi
+
+    ${CODE_ROOT}/${cesmtag}/cime/scripts/create_newcase \
+        --case ${CASEROOT}  \
+        --compset ${compset} \
+	--res ${resolution} \
+        --run-unsupported  \
+	--project ${project} 
+	#--workflow cmip7
+fi
+
+########################
+## copy script to CASEROOT directory
+########################
+SCRIPT_PATH="$(readlink -f "$0")"
+SCRIPT_DIR="$(dirname "$SCRIPT_PATH")"
+SCRIPT_NAME="$(basename "$SCRIPT_PATH")"
+DATESTAMP="$(date +%Y%m%d)"
+USE_NAME="${SCRIPT_NAME}-${DATESTAMP}"
+
+mkdir -p "${CASEROOT}/provenance_script"
+cp "$SCRIPT_PATH" "${CASEROOT}/provenance_script/$USE_NAME"
+
+########################
+## add to github repo
+########################
+if [[ $do_git_archive == true ]]; then
+    cd ${CASEROOT}  
+    cp $curdir
+    ./xmlchange CASE_GIT_REPOSITORY=git@github.com:NCAR/cesm_dev.git
```

Comment:
You could add the bit about archiving the script that runs the case here with something like:

``` shell
cp $0 .
git add $0
git commit -m "Archive the bash script that created this case" $0
```

I think that should work to both archive it initially, and to archive updates to it if the script is run again.

Reply:
@jedwards4b would know more about this -- but it seems like you'll need to add some points in the script where it does a git commit to save new changes. And I assume it'll need some "git push" commands after them as well.

Reply:
This is already done in cime, no need for git commands here.

Reply:
Thanks Erik and Jim.  Let me know if you still want me to change anything (I've moved the section down)